### PR TITLE
add CAPZ v1.14 to the prow plugins

### PR DESCRIPTION
**Pull request by nawazkh**
/assign @CecileRobertMichon @jackfrancis

### CODE REVIEW

File: config/prow/plugins.yaml [hunk 1]
```diff
@@ -513,7 +513,8 @@ milestone_applier:
     release-0.7: v0.7.x
     release-0.6: v0.6.x
   kubernetes-sigs/cluster-api-provider-azure:
-    main: v1.13
+    main: v1.14
+    release-1.13: v1.13
```
Comment (nawazkh) [inline config/prow/plugins.yaml:517]: Other providers seem to be maintaining a list of their release, where as we are only maintaining the recent two.
Are we doing this to honor the release support of recent two releases or is there more to it?

Comment (nawazkh) [reply config/prow/plugins.yaml:517]: Had an adhoc discussion on this: The listed branches are the supported branches at CAPZ.
Will remove `release-1.11`

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**